### PR TITLE
ChordDetection fixes

### PR DESCRIPTION
**Pull request by seffka**

### CODE REVIEW

File: src/algorithms/tonal/chordsdetectionbeats.cpp [hunk 1]
```diff
@@ -42,6 +42,9 @@ const char* ChordsDetectionBeats::description = DOC("This algorithm estimates ch
 void ChordsDetectionBeats::configure() {
   _sampleRate = parameter("sampleRate").toReal();
   _hopSize = parameter("hopSize").toInt();
+  _chromaPick = parameter("chromaPick").toLower();
+  if (!(_chromaPick == "interbeat_median" || _chromaPick == "first_beat"))
+    throw EssentiaException("Bad DCT type.");
```
Comment (dbogdanov) [inline src/algorithms/tonal/chordsdetectionbeats.cpp]: Throw a more informative message: 
```throw EssentiaException("ChordsDetectionBeats: Bad 'chromaPick' parameter");```

File: src/algorithms/tonal/chordsdetectionbeats.h [hunk 2]
```diff
@@ -54,6 +55,7 @@ class ChordsDetectionBeats : public Algorithm {
     void declareParameters() {
       declareParameter("sampleRate", "the sampling rate of the audio signal [Hz]", "(0,inf)", 44100.);
       declareParameter("hopSize", "the hop size with which the input PCPs were computed", "(0,inf)", 2048);
+      declareParameter("chromaPick", "method of calculating singleton chroma for interbeat interval", "{first_beat,interbeat_median}", "interbeat_median");
```
Comment (dbogdanov) [inline src/algorithms/tonal/chordsdetectionbeats.h]: We'll need a better explanation of what ```first_peak``` and  ```interbeat_median``` do. Expand the DOC string in the cpp file with a more detailed description.

Comment (seffka) [reply src/algorithms/tonal/chordsdetectionbeats.h]: Done. Could you please check?

File: src/algorithms/tonal/chordsdetectionbeats.cpp [hunk 3]
```diff
@@ -27,7 +27,12 @@ namespace standard {
 
 const char* ChordsDetectionBeats::name = "ChordsDetectionBeats";
 const char* ChordsDetectionBeats::category = "Tonal";
-const char* ChordsDetectionBeats::description = DOC("This algorithm estimates chords using pitch profile classes on segments between beats. It is similar to ChordsDetection algorithm, but the chords are estimated on audio segments between each pair of consecutive beats.\n"
+const char* ChordsDetectionBeats::description = DOC(
+"This algorithm estimates chords using pitch profile classes on segments between beats. "
+"It is similar to ChordsDetection algorithm, but the chords are estimated on audio segments between each pair "
+"of consecutive beats. Chroma vector from the segment could be obtained as:\n"
+"  - 'interbeat_median', each resulting chroma vector component is a median of all the component values in the segment\n"
+"  - 'first_beat', chroma vector is sampled from the segment's first beat (makes sense if chroma is preliminary smoothed)\n"
```
Comment (dbogdanov) [inline src/algorithms/tonal/chordsdetectionbeats.cpp]: All segment are covering audio between two consecutive beats. The "segment's starting beat" would be more clear. Maybe something like this? "Chroma vector is sampled from the start of each segment (that is, its starting beat position) using the first frame in each segment."

Comment (seffka) [reply src/algorithms/tonal/chordsdetectionbeats.cpp]: Ok, It's better. I use this phrasing and rename first_beat to starting_beat (committed)

File: src/algorithms/tonal/chordsdetectionbeats.cpp [hunk 4]
```diff
@@ -48,7 +48,7 @@ void ChordsDetectionBeats::configure() {
   _sampleRate = parameter("sampleRate").toReal();
   _hopSize = parameter("hopSize").toInt();
   _chromaPick = parameter("chromaPick").toLower();
-  if (!(_chromaPick == "interbeat_median" || _chromaPick == "first_beat"))
```
Comment (dbogdanov) [inline src/algorithms/tonal/chordsdetectionbeats.cpp]: Need to update the parameter in the header file too.

Comment (seffka) [reply src/algorithms/tonal/chordsdetectionbeats.cpp]: Ok, done.

**Comment by dbogdanov** [comment]
Thanks!

**Pull request merged**